sdk/python/kfp/_client.py

```diff
@@ -52,21 +55,23 @@ def __init__(self, host=None, client_id=None, namespace='kubeflow'):
     """
 
     self._host = host
+    self._uihost = os.environ.get(KF_PIPELINES_UI_ENDPOINT_ENV, '')
```

Comment:
Can default value be host? Otherwise, it will break currently local notebook + port forward use case.

Reply:
@hongye-sun i'm trying to make the UI and API ports configurable from the environment variables, this way we dont have to use proxy server when there is a proper API/UI gateway.

i wasnt clear about the port forward case, are you forwarding the UI port or the API port, currently the it seems like `host` and `config.host` identifies the API port and the current code is broken, im one place you have:

```
     if in_cluster:
      config.host = Client.IN_CLUSTER_DNS_NAME.format(namespace)
      return config
```
(i.e. host == API port)

in another:
```
    if config.host:
      config.host = os.path.join(config.host, Client.KUBE_PROXY_PATH.format(namespace))
```

Reply:
@yaronha, it makes sense to have API and UI addresses as separate configurations. We have a wiki to guide user to local port-forward on ambassador (https://www.kubeflow.org/docs/other-guides/accessing-uis/), which is the gateway for both API and UIs.  We also have a notebook tutorial (https://github.com/amygdala/examples/blob/cookbook/cookbook/pipelines/notebooks/kfp_remote_deploy-no-IAP.ipynb) to guide user to use the SDK with local port forward and I think the UI link is working at this time. IAP guide (https://github.com/amygdala/examples/blob/cookbook/cookbook/pipelines/notebooks/kfp_remote_deploy.ipynb) should probably work, too. I think we should make the existing tutorial continues working, which assumes API and UI are both from the same host. That's why I suggest to keep the host as uihost if the env var is missing.

Reply:
@hongye-sun the examples above fall into two categories, using Proxy and using proprietary Google API gateway (IAP),  we or others may have our own UI/API gateway with strong identity management (like IAP) so need a way to specify the the path to it.

The API and UI can be different hosts, since the API can be an internal cluster/DNS name, while the UI is accessed from a remote browser and need a publicly exposed URL.

Reply:
@yaronha, thanks for letting me know your use cases. let's keep both host and uihost. The only thing missing in the PR is that it changes the behavior for uihost if user doesn't provide env var, which might cause existing tutorial break. Could you make sure that in case of env var is missing, the uihost equals to host, just like today?

Reply:
Thanks, i will update

Reply:
@hongye-sun did u review my changes?

Reply:
Sorry for the delay. One more comment.